lib/IRGen/IRGenSIL.cpp

```diff
@@ -2487,6 +2538,12 @@ void IRGenModule::emitSILFunction(SILFunction *f) {
       f->isAvailableExternally())
     return;
 
+  // Type metadata for foreign references is not yet supported on Windows. Bug #76168.
```

Comment:
Is there a reason that this is not supported on Windows?

Reply:
My understanding is that on Windows we end up requiring a metadata completion function for C++ foreign reference types. That logic just isn't implemented yet for C++ types.

https://github.com/swiftlang/swift/blob/bd313deae5fef61d7e167a90f5dd39a2dea4b251/lib/IRGen/GenMeta.cpp#L6636-L6639